test/integration/model/create.test.js

```diff
@@ -330,6 +330,16 @@ describe(Support.getTestDialectTeaser('Model'), () => {
       });
     });
 
+    it('returns instance if username is a sequelize fn with $ in value', function() {
+      const data = {
+        username: Sequelize.fn('upper', '$username')
+      };
+
+      return this.User.create(data).then(user => {
+        expect(user.username).to.equal('$USERNAME');
```

Comment:
@papb this test is failing for mysql and mariadb, is this some other bug or those dialect doesn't support functions